netplan_cli/cli/state_diff.py

```diff
@@ -442,6 +486,62 @@ def _normalize_routes(self, routes: set) -> set:
 
         return new_routes_set
 
+    def _normalize_gateways(self, config: dict) -> AbstractSet[NetplanRoute]:
+        ''' Convert the gateway4 and gateway6 properties to NetplanRoutes.
+        The only information stored by these properties is the destination IP for
+        the default route. Networkd and Network Manager will set a default metric
+        for it when the route is installed. As we have no control over it, there
+        will always be a diff. The same is true for the route table number.
+        To avoid the noise in the diff report when these properties are used,
+        we use the following heuristic: if there is one and only one static default
+        route in the system and gateway4 and/or gateway6 are used, we set the same
+        metric and table as found in the system.
+        '''
+        routes_set = set()
+        system_interfaces = self.system_state.get_data()
+        interface = config.get('id')
+
+        ifaces = [iface for iface in system_interfaces.values()
+                  if iface.get('id') == interface]
+        routes = [route for iface in ifaces for route in iface.get('routes', [])]
+
+        if gateway4 := config.get('gateway4'):
+            default_routes = [route for route in routes
+                              if route.get('to') == 'default'
```

Comment:
question (non-blocking): Are we sure this is already normalized to `default`, or could it also be something like `0.0.0.0/0`?